### DISCUSSION TITLE
Pascal-170 vs Pascal-6000

### DISCUSSION THREAD

**Discussion by KC5VDJ** [kej715/DtCyber#118] [General]
Other than access to the SCREEN Panels UI, and FTN5, was there any advantage to Pascal-170?  It seems to me to be a very deficient subset by comparison.  Did it do better code generation for 170s?  Interesting that Wirth did some actual coding in '6000's libs.

Just wondering, although it was three-pass, was Oregon Software's ISO for PDP-11 based on ETH as well, or was it a true Indy?

One thing I never got (like basic "let") was the need for "array" in a type definition.  The first '[' implies an array.

PS: He missed being 90 by six weeks.

RIP Prof. Wirth, 01/01/2024.  He was a strange, brilliant man.  In an endless battle against complexity and crap code, creating beautiful languages that require code generation of ever increasing complexity in order for all of the bounds and sanity checking at runtime to occur transparently to the programmer... 😁 A legend has died.  Woulda loved to see a requiem from Edgar Dijkstra, but he died two decades ago...

**Comment by kej715** [comment]
Assuming that "Pascal-170" is the CDC-provided compiler, and "Pascal 6000"
is the U of Zurich / U of MN compiler ... the heritage of Pascal-170 is not
clear, but both compilers probably have a common ancestor, the original
compiler implemented by Wirth. CDC may have licensed an early version and
derived Pascal-170 from that. CDC invested minimally in it. Meanwhile, the
Universities of Zurich and Minnesota continued developing the compiler. For
example, Pascal 4 supports mixed case, and the U of Minnesota implemented a
very nice system interface library for it. Both the Cray Station software
and the UMass Mailer require Pascal 4 (aka Pascal 6000 v 4).

Both compilers are available in NOS 2.8.7. The base system includes
Pascal-170. If the product named "pascal4-pf" is added to the base system,
Pascal 4 will be installed as a set of permanent files at UN=LIBRARY, and
Pascal-170 will remain in the base system library. You can then compare
their performance and code generation side-by-side.

**Comment by KC5VDJ** [comment]
Thanks!

Pascal was the first usable language I learned (basic / 6502 asm, first), in high school on the 11/44.  I suspect a conspiracy of teachers "stole" the pdp-8 a year or two before; if true, they deserve a medal; teaching kids a pdp-8 in the mid 80's was kinda nuts, because even in control apps, other stuff (PCs and STD / MULTIBUS / VME) had taken over the most of the pdp-8 space...  Hell at that point, VCRs used Z-80s and (usually) Hitachi 68HC-series controllers.

I can say one thing that gen of DEC kinda imparted on people was that terminals didn't have to be clunky; the vt-220 was a game changer in form factor, and the LK-201 quite usable, even though membrane (compare to the sadist-designed HP terminal keyboards which lasted into this century).
